### ISSUE TITLE
LAN Only Fails

### ISSUE THREAD

**Issue by cheme75** [mrlt8/docker-wyze-bridge#389]
Would like to get this working to ease data usage, got 3 cams using about 1gb/hr in Relay mode. I’m guessing it’s a matter of opening the right ports so the bridge can read from the cams directly. Or is it more complicated?

**Comment by mrlt8** [comment]
Could this be a firewall issue? It should find the cameras and connect directly if they're all on the same network.

**Comment by cheme75** [comment]
I’m guessing it is firewall related, all are on same subnet… only got the one. Just don’t know what port I might need to open. I had to open 8888 and 8554 so I can view on browser and get the rtsp into homebridge/ffmpeg or scrypted. Nor sure what I’m missing.

**Comment by cheme75** [comment]
I tried again and all I got were failures - not sure what I need to open or allow on firewall? very frustrating. do I need to allow each cam's IP for some protocol?

**Comment by cheme75** [comment]
@mrlt8 - Here is log when I change to LAN mode - I have ports 8888, 8554, 1935 allowed on LAN - What am I missing? Do I need to create a rule for each cam IP in the windows firewall? 

![image](https://user-images.githubusercontent.com/16415255/170281231-999f1721-39a7-4a20-929a-b3560fd05724.png)

**Comment by justinj57** [comment]
> @mrlt8 - Here is log when I change to LAN mode - I have ports 8888, 8554, 1935 allowed on LAN - What am I missing? Do I need to create a rule for each cam IP in the windows firewall?
> 
> ![image](https://user-images.githubusercontent.com/16415255/170281231-999f1721-39a7-4a20-929a-b3560fd05724.png)

You've opened the ports the bridge uses to transmit the feeds, but it uses other ports to connect to the cams. Can you do a packet capture on the IP of one of the cameras when you fire up the bridge container?

**Comment by cheme75** [comment]
I got same issue! I allowed UDP on the pc to/from all on local subnet, but still fails. Not sure what specific ports to open. Not sure what app to use to capture packets?  Support for this issue has been weak. I just stop my bridge overnight since it’s guard gated so we get extremely limited traffic at night. This at least cuts data usage in half. But should be able to run it on lan only.

**Comment by mrlt8** [comment]
Can you temporarily disable the firewall on windows or try to test run it on another machine?

**Comment by justinj57** [comment]
> I got same issue! I allowed UDP on the pc to/from all on local subnet, but still fails. Not sure what specific ports to open. Not sure what app to use to capture packets? Support for this issue has been weak. I just stop my bridge overnight since it’s guard gated so we get extremely limited traffic at night. This at least cuts data usage in half. But should be able to run it on lan only.

It's a free product developed in someone's spare time, can't really expect 24/7 support. Especially considering this is likely an issue with your network architecture, not with the project...

What router are you running and what OS is your docker host running?

**Comment by cheme75** [comment]
> 

I had some issue trying to turn off firewall, thing machine still has some group policy issues. I’ll retry tomorrow. It’s win 10 pro. Don’t think it’s a router issue. I’ve been able to run homebridge and scrypted with remote access their UI over the lan and I’ve opened ports for for crypto node, my printer and NAS and all are working fine inside and remotely.Fwiw -  It’s the arris bgw210 from att. Using it for now since it’s been working ok.

**Comment by justinj57** [comment]
> > 
> 
> I had some issue trying to turn off firewall, thing machine still has some group policy issues. I’ll retry tomorrow. It’s win 10 pro. Don’t think it’s a router issue. I’ve been able to run homebridge and scrypted with remote access their UI over the lan and I’ve opened ports for for crypto node, my printer and NAS and all are working fine inside and remotely.Fwiw - It’s the arris bgw210 from att. Using it for now since it’s been working ok.

I believe the tutk library uses a randomly generated UDP port to establish a connection, I would try to get the firewall disabled again and see if you're still having problems.

**Comment by cheme75** [comment]
> 

I spent half the night on my pc using the admin acct and finally found some places in registry to disable firewall. Unfortunately it did not change anything, LAN only still fails to connect. So it must be the firewall restricting UDP traffic. Not sure why it would do so on the local LAN. I’ll see if I can find anything on the router tomorrow.

**Comment by cheme75** [comment]
> Any chance you have spare raspberry or other machine you could test the bridge on to see if it's a network or windows issue?

Only spare I got remaining is a 12 yr old second gen i5 “win 10 home” laptop I haven’t used in quite some time. I was under impression this needed a win pro version. 

I guess even with the firewall down, there may be something stuck somewhere in the registry/group policy maybe. Why back this hard drive was in a corporate win 7 laptop that I used to test win10 before upgrading my home office. I moved this drive to my old desktop which was without a drive and reinstalled win10 pro from my personal account in order to run docker for a crypto test node. Since it’s just temporary, I added this bridge and it’s been doing great - only glitch is no LAN mode. It’s a bit frustrating - considering wiping windows and going to some Linux distro, but I haven’t used Linux much since I had a CentOS server 15 yrs ago. 

Fwiw, I went through the router and could see the 3 cams ip are all using a TCP port and various UDP ports, but I’m not seeing any signs any of those ports are being blocked. I haven’t yet checked to see who the IPs belong to, suspect AWS. I just need to double check the router logs to see what has been blocked just to be sure.

**Comment by mrlt8** [comment]
https://docs.docker.com/desktop/windows/install/ says "Windows 10 64-bit: Home or Pro 21H1 (build 19043) or higher"
so I assume home edition should work.

**Comment by cheme75** [comment]
Oh, ok. Maybe it’s the 64 that threw me. All mine are 64 so I’ll fire up laptop and get it fully updated then load up docker to try it for the bridge. Might take me some time between other things I got going.

**Comment by cheme75** [comment]
@mrlt8 @justinj57 - took a while to update the old laptop, but finally got docker to run, got bridge pulled and up now and sure enough it’s using lan mode by default. That old hd from a corp laptop in the desktop must have domain/sys admin policies screwing it up. I’ll have to see if I can find the policies to clear out. One thing I noticed is this docker install asked to add firewall rules, I’ll have to see if I can find them and maybe can set same on the desktop. Any suggestions on deleting old policies/possible restrictive firewall rules will be appreciated.

**Comment by mrlt8** [comment]
Sorry, I don't really know much about windows. Could you try upgrading/re-installing docker-desktop?

**Comment by cheme75** [comment]
Yup, done that, not fix yet. I did manage to find all the group policies, deleted them and got firewall off. Compared network settings between the two and see nothing yet that would be different. Desktop is wired Ethernet, laptop is WiFi. Gotta find a long enough cord to switch laptop to wired just to be sure it’s not interface related, but I can’t see why since it’s all on same router. Before I shut off firewall, I did check the docker firewall rules and made sure it had all tcp and all udp ports allowed. But that did not help. Im guessing it’s something beyond the firewall in the registry. Would be nice if I could wipe out the firewall settings including the registry settings and copy them from the laptop.

**Comment by hmuscroft** [comment]
In case it helps, I have 4 x Wyze v3 cams running on firmware 4.61.0.3 which was the BETA firmware Wyze released with RTSP support (which was ropey at best). I decided to try this Docker solution instead and it worked much better. I then thought I would return to the latest "RELEASE" firmware version (4.36.9.139) so I flashed one of the cameras back to that. Immediately that camera would failed to connect via LAN and would only connect via P2P.

I wonder whether the firmware version dictates whether a LAN connection can be made to the Wyze Cam v3?

**Comment by Doghaus96** [comment]
wanted to chime in and help if at all possible (absolutely love this project). Running into the same issue as OP, if I set it to LAN only it fails every time, if its unset it uses relay. Running on a raspi 4, connected via wifi, using a docker-compose to bring it up. 1 wyze cam v2, wyze pan, wyze cam v3, and doorbell v1 (stuck on the fubared fw).  If there's anything I can try to be helpful let me know!

**Comment by cheme75** [comment]
Ok, I’m on LAN mode now - long story, will try to shorten - found my win machine locked up, restart went into disc repair. After this I did a major cleanup and must have broken something since docker refused to run. Possibly deep cleaning deleted a necessary system file. Anyway, figured good opportunity to reboot to disc utility to delete an unused partition and expand the C drive, not huge but 15gb was wasted for old laptop mfg restore partition, so gained 15gb, tested hd for errors, looked good, rebooted. No luck on docker. Uninstalled docker. Rebooted, reinstalled docker, rebooted, opened dashboard and docker was updated my other container. Ran docker-compose pull from my bridge folder, then up -d and… tada, it’s now defaulting to LAN mode! Didn’t even need to force it in yml file. So far it’s looking good. Must be orig docker install had some glitch somewhere. Not sure on what exactly since I didn’t change a thing in the firewall settings. Possibly editing firewall should be done prior to installing docker. Originally I had already installed docker then edited the firewall, I did reboot several times for various reasons, but no reboot ever changed things, was stuck in relay mode. Whatever reason, it’s in LAN mode and all is looking good.
